### PR TITLE
KCAL-28: Maybe record_call when retrieving from parking

### PR DESCRIPTION
**Pull request by harenson**
5.0: https://github.com/2600hz/kazoo-callflow/pull/86

Related to https://github.com/2600hz/kazoo-crossbar/pull/215

### CODE REVIEW

File: src/module/cf_park.erl [hunk 1]
```diff
@@ -118,10 +118,61 @@ handle_nomatch_with_empty_referred_to(Data, Call, PresenceType, ParkedCalls, Slo
...
+%% - If device has call_recording configuration and call recording is disabled, do not record call.
+%%   If it is enabled, then record the call. Same logic for user up to account.
+%% @end
+%%------------------------------------------------------------------------------
+-spec maybe_record_call(kapps_call:call(), kz_json:object(), kz_json:object(), kz_json:object()) -> boolean().
+maybe_record_call(Call, EndpointRec, UserRec, AccountRec) ->
+    Key = fun(Dir) -> [Dir, kapps_call:inception_type(Call), <<"enabled">>] end,
+    AnyKey = Key(<<"any">>), %% any
+    DirKey = Key(kapps_call:direction(Call)), %% inbound, outbound
+    IsEnabled = fun(RecJObj) ->
```
Comment (jamesaimonetti) [inline src/module/cf_park.erl]: What does this `fun()` gain the `maybe_record_call/4` function's read-ability versus defining `is_recording_enabled(AnyKey, DirKey, RecJObj) -> kz_json:get_first_defined([AnyKey, DirKey], RecJObj).` ?

File: src/module/cf_park.erl [hunk 2]
```diff
@@ -118,10 +118,62 @@ handle_nomatch_with_empty_referred_to(Data, Call, PresenceType, ParkedCalls, Slo
...
+%% For example:
+%% - If device is NOT configured to record calls, check user, if user is NOT configured to record
+%%   calls, check account.
+%% - If device has call_recording configuration and call recording is disabled, do not record call.
+%%   If it is enabled, then record the call. Same logic for user up to account.
+%% @end
+%%------------------------------------------------------------------------------
+-spec maybe_record_call(kapps_call:call(), kz_json:object(), kz_json:object(), kz_json:object()) -> boolean().
+maybe_record_call(Call, EndpointRec, UserRec, AccountRec) ->
+    Key = fun(Dir) -> [Dir, kapps_call:inception_type(Call), <<"enabled">>] end,
```
Comment (jamesaimonetti) [inline src/module/cf_park.erl]: I really don't see the point of a function here.

Why not `AnyKey = [<<"any">>, kapps_call:inception_type(Call), <<"enabled">>],` directly?

File: src/module/cf_park.erl [hunk 3]
```diff
@@ -118,10 +118,59 @@ handle_nomatch_with_empty_referred_to(Data, Call, PresenceType, ParkedCalls, Slo
...
+    DirKey = [kapps_call:direction(Call), InceptionType, <<"enabled">>], %% inbound, outbound
+    maybe_record_call(RecJObjs, AnyKey, DirKey).
+
+maybe_record_call([RecJObj | RecJObjs], AnyKey, DirKey) ->
+    case is_recording_enabled(AnyKey, DirKey, RecJObj) of
+        'undefined' -> maybe_record_call(RecJObjs, AnyKey, DirKey);
+        MaybeEnabled -> kz_term:is_true(MaybeEnabled)
+    end.
+
+-spec is_recording_enabled([kz_term:ne_binary(), ...], kz_term:ne_binary(), kz_json:object()) -> boolean().
```
Comment (jamesaimonetti) [inline src/module/cf_park.erl]: You blew away my spec fixes in 87cab49397831ed77c28da9fbcce3b3ea453d7ab LOL

AnyKey and DirKey should be `kz_json:path()` and the function should return `kz_term:api_boolean()`

Comment (harenson) [reply src/module/cf_park.erl]: And that's why I had to force push my changes 🤦🏾

**Pull request merged**